lightly/api/upload.py

```diff
@@ -39,121 +37,15 @@
-                               dataset_id: str,
-                               token: str,
-                               max_upload: int = 32,
-                               embedding_name: str = 'default',
-                               verbose: bool = True):
-    """Uploads embeddings from a csv file to the cloud solution.
-
-    The csv file should be in the format specified by lightly. See the
-    documentation on lightly.utils.io for more information.
-
-    Args:
-        path_to_embeddings:
-            Path to csv file containing embeddings.
-        dataset_id:
-            The unique identifier for the dataset.
-        token:
-            Token for authentication.
-        max_upload:
-            Size of a batch of embeddings which is uploaded at once.
-        embedding_name:
-            Name the embedding will have on the Lightly web-app.
-
-    Raises:
-        RuntimeError if there is an error during the upload
-        of a batch of embeddings.
-
-    """
-    data, embeddings, _, _ = load_embeddings_as_dict(
-        path_to_embeddings,
-        embedding_name,
-        return_all=True,
-    )
-
-    data['token'] = token
-    data['datasetId'] = dataset_id
 
-    tags = routes.users.datasets.tags.get(dataset_id, token)
-    if len(tags) == 0:
-        msg = 'Forbidden upload to dataset with no existing tags.'
-        raise RuntimeError(msg)
-
-    embedding_summaries = routes.users.datasets.embeddings.get_summaries(
-        dataset_id, token)
-    embedding_names = [embedding['name'] for embedding in embedding_summaries]
-    if embedding_name in embedding_names:
-        msg = f'Forbidden upload: Embedding name {embedding_name} '
-        msg += 'already exists.'
-        raise RuntimeError(msg)
-
-    # use pca to make 2d embeddings
-    transformer = fit_pca(embeddings)
-    n_embeddings = len(data['embeddings'])
-    n_batches = n_embeddings // max_upload
-    n_batches = n_batches + 1 if n_embeddings % max_upload else n_batches
-
-    embedding_batches = [None] * n_batches
-    for i in range(n_batches):
-        left = i*max_upload
-        right = min((i + 1) * max_upload, n_embeddings)
-        batch = data.copy()
-        batch['embeddings'] = data['embeddings'][left:right]
-        embedding_batches[i] = batch
-
-    if verbose:
-        print('Uploading embeddings:')
-
-    pbar = tqdm.tqdm(unit='embs', total=n_embeddings)
-    for i, batch in enumerate(embedding_batches):
-        _upload_single_batch(
-            dataset_id, token, batch, transformer, append=i > 0)
-        pbar.update(len(batch['embeddings']))
+def get_api_client(token: str, host: str = None) -> ApiClient:
+    if host is None:
+        host = getenv('LIGHTLY_SERVER_LOCATION', 'https://api.lightly.ai')
+    configuration = Configuration()
+    configuration.host = host
+    configuration.api_key = {'token': token}
+    api_client = ApiClient(configuration=configuration)
+    return api_client
 
 
 def _upload_single_image(image,
```

Comment:
Would it not be more consistent to have these functions in `api_workflow_upload_dataset.py` like you did it with the other workflows?

Reply:
Yes, I did that